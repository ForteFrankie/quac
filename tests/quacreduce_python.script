#!/bin/bash

# Test that the Python interface of QUACreduce works.

. ./environment.sh

cd $DATADIR


## Set up input

y "echo -e 'foo bar baz\nfoo foo' > foo1.txt"
y "echo -e 'bar' > foo2.txt"
y "cat foo*.txt"


## Do map-reduce

y "quacreduce --python qr.wordcount.Job ./job foo*.txt"
y "cd job && make --quiet"  # output contains temp dirs that vary between tests
y "cat job/out/* | sort"


## Clean up for subsequent tests

rm -Rf job
