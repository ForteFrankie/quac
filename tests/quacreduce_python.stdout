$ (echo -e 'foo bar baz\nfoo foo' > foo1.txt)
$ (echo -e 'bar' > foo2.txt)
$ (cat foo*.txt)
foo bar baz
foo foo
bar
$ (quacreduce --python qr.wordcount.Job ./job foo*.txt)
$ (cd job && make --quiet)
$ (cat job/out/* | sort)
1 baz
2 bar
3 foo
